packages/mobile/src/transactions/feed/TransactionDetailsScreen.tsx

```diff
@@ -0,0 +1,119 @@
+import { tokensByCurrencySelector } from 'src/tokens/selectors'
+import TransferSentContent from 'src/transactions/feed/detailContent/TransferSentContent'
+import {
+  TokenExchange,
+  TokenTransaction,
+  TokenTransactionTypeV2,
+  TokenTransfer,
+} from 'src/transactions/types'
+import { Currency } from 'src/utils/currencies'
+import { getDatetimeDisplayString } from 'src/utils/time'
+import RewardReceivedContent from './detailContent/RewardReceivedContent'
+import TransferReceivedContent from './detailContent/TransferReceivedContent'
+
+type Props = StackScreenProps<StackParamList, Screens.TransactionDetailsScreen>
+
+function useHeaderTitle(transaction: TokenTransaction) {
+  const { t } = useTranslation()
+  const tokensByCurrency = useSelector(tokensByCurrencySelector)
+  const celoAddress = tokensByCurrency[Currency.Celo]?.address
+  const rewardsSenders = useSelector(rewardsSendersSelector)
+  const addressToDisplayName = useSelector(addressToDisplayNameSelector)
+
+  switch (transaction.type) {
+    case TokenTransactionTypeV2.Exchange:
+      // TODO: Change this to show any exchanges, not just CELO sell/purchase.
+      const isCeloSell = (transaction as TokenExchange).inAmount.tokenAddress === celoAddress
+      return isCeloSell ? i18n.t('soldGold') : i18n.t('purchasedGold')
+    case TokenTransactionTypeV2.Sent:
+      const isCeloSend = (transaction as TokenTransfer).amount.tokenAddress === celoAddress
+      return t(isCeloSend ? 'transactionHeaderWithdrewCelo' : 'transactionHeaderSent')
+    case TokenTransactionTypeV2.Received:
+      const transfer = transaction as TokenTransfer
+      const isCeloReception = transfer.amount.tokenAddress === celoAddress
+      if (
+        rewardsSenders.includes(transfer.address) ||
+        addressToDisplayName[transfer.address]?.isCeloRewardSender
+      ) {
+        return t('transactionHeaderCeloReward')
+      } else {
+        return isCeloReception
+          ? i18n.t('transactionHeaderCeloDeposit')
+          : i18n.t('transactionHeaderReceived')
+      }
+    case TokenTransactionTypeV2.InviteSent:
+      return t('transactionHeaderEscrowSent')
+    case TokenTransactionTypeV2.InviteReceived:
+      return t('transactionHeaderEscrowReceived')
+  }
+}
+
+function TransactionDetailsScreen({ navigation, route }: Props) {
+  const { transaction } = route.params
+
+  const headerTitle = useHeaderTitle(transaction)
+  useLayoutEffect(() => {
+    const dateTimeStatus = getDatetimeDisplayString(transaction.timestamp, i18n)
+    navigation.setOptions({
+      headerTitle: () => <HeaderTitleWithSubtitle title={headerTitle} subTitle={dateTimeStatus} />,
+    })
+  }, [transaction])
+
+  const addressToDisplayName = useSelector(addressToDisplayNameSelector)
+  const rewardsSenders = useSelector(rewardsSendersSelector)
+
+  let content
+
+  switch (transaction.type) {
+    case TokenTransactionTypeV2.Exchange:
+      // TODO:
+      content = null
```

Comment:
Will be in a follow up PR?

Reply:
Yes, the issue is here: https://app.zenhub.com/workspaces/retention-squad-sprint-board-612f2ef98ef2b800116e6043/issues/valora-inc/wallet/1595